cmake/GtsamCythonWrap.cmake

```diff
@@ -5,13 +5,17 @@ unset(PYTHON_EXECUTABLE CACHE)
 unset(CYTHON_EXECUTABLE CACHE)
 unset(PYTHON_INCLUDE_DIR CACHE)
 unset(PYTHON_MAJOR_VERSION CACHE)
+unset(PYTHON_LIBRARY CACHE)
```

Comment:
GTSAM_USE_CUSTOM_PYTHON_LIBRARY?

Reply:
This is to unset the cache for PYTHON_LIBRARY (libpython paths)